processing/src/main/java/org/apache/druid/query/filter/InDimFilter.java

```diff
@@ -149,28 +153,36 @@ public FilterTuning getFilterTuning()
   @Override
   public byte[] getCacheKey()
   {
+    if (valuesHashCode == null) {
+      valuesHashCode = computeValuesHashCode(values);
+    }
     if (cacheKey == null) {
-      final List<String> sortedValues = new ArrayList<>(values);
-      sortedValues.sort(Comparator.nullsFirst(Ordering.natural()));
-      final Hasher hasher = Hashing.sha256().newHasher();
-      for (String v : sortedValues) {
-        if (v == null) {
-          hasher.putInt(0);
-        } else {
-          hasher.putString(v, StandardCharsets.UTF_8);
-        }
-      }
       cacheKey = new CacheKeyBuilder(DimFilterUtils.IN_CACHE_ID)
           .appendString(dimension)
           .appendByte(DimFilterUtils.STRING_SEPARATOR)
           .appendByteArray(extractionFn == null ? new byte[0] : extractionFn.getCacheKey())
           .appendByte(DimFilterUtils.STRING_SEPARATOR)
-          .appendByteArray(hasher.hash().asBytes())
+          .appendByteArray(valuesHashCode.asBytes())
           .build();
     }
     return cacheKey;
   }
 
+  private static HashCode computeValuesHashCode(Set<String> values)
+  {
+    final List<String> sortedValues = new ArrayList<>(values);
+    sortedValues.sort(Comparator.nullsFirst(Ordering.natural()));
+    final Hasher hasher = Hashing.sha256().newHasher();
```

Comment:
I know this isn't new, but might be worth investigating if there are faster hashes that are unique enough for this (not necessary in this PR, just thinking out loud)